corehq/ex-submodules/pillowtop/processors/form.py

```diff
@@ -77,13 +77,7 @@ def _last_submission_needs_update(last_submission, received_on_datetime, build_v
     time_difference = received_on_datetime - last_submission.submission_date
     debounce_delay = settings.USER_REPORTING_METADATA_UPDATE_FREQUENCY
     update_frequency = timedelta(minutes=debounce_delay) if debounce else timedelta(seconds=0)
-    if time_difference > update_frequency:
-        return True
-    if build_version != last_submission.build_version:
-        return True
-    if cc_version != last_submission.commcare_version:
-        return True
```

Comment:
The failure mode / thing that this change had intended to provide is allowing app updates to appear faster than the normal `update_frequency` yeah? Are we worried about losing that behavior?

And if so, are our version numbers such that we could use `>=` instead of `!=`?

Also do we not have `received_on` dates we can compare?

Reply:
For the last question I think if you add `received_on_datetime > last_submission.submission_date` to all the checks that would also solve it, yeah?

Reply:
Thanks for the added context, I couldn't quite figure out what the failure mode it was trying to solve was. That makes sense that that was the intended function of this code. Though in practice I would be surprised if these conditions are hit very frequently.

We could use >= for build version, but I think we'd need to use something like `LooseVersion` for commcare version. I didn't change that here because != would support rollbacks or a user using multiple devices with different versions. Not that those are important workflows that we need to consider.

I solved a it differently than you suggested, but I believe it has the intended behavior https://github.com/dimagi/commcare-hq/pull/25639/commits/3489302bb7fbbac23e8fc1477e1d79663a2e757f

Reply:
I just guessed on the context, but solution looks good.